### ISSUE TITLE
Navigation tree collapse if page link's anchor is unicode text.

### ISSUE THREAD

**Issue by 0giso** [rtfd/sphinx_rtd_theme#495]
Sphinx with recommonmark does not use unicode text header as page link's anchor, just ignore them. If the header is written by unicode text only, anchor is empty like "http://example.com/foo.html#". So I use [unicode_ids extension](https://pypi.python.org/pypi/unicode_ids) to generate unicode anchor text from header.

In that case, navigation to 3rd or below level get fall back to top level content and toc tree collapsed, because sphinx_rtd_theme try to find URI escaped anchor and fail.

Change theme.js from
`var anchor = encodeURI(window.location.hash);`
to
`var anchor = window.location.hash;`
worked for me. Is this acceptable change?

**Comment by Blendify** [comment]
@agjohnson you are the javascript guy

**Comment by jun-ogiso** [comment]
> var anchor = window.location.hash;

Sorry, that worked only with Edge and IE.

This may be just a workaround, but the code below works with Edge, IE, Chrome.
`var anchor = decodeURI(window.location.hash);`

**Comment by remyabel** [comment]
Yes, I think that `decodeURI` is more correct than `encodeURI`. For the test string `ёђѓєѕіїјљњћќўџабвг`, the value of `window.location.hash` is:

- Firefox: `%D1%91%D1%92%D1%93%D1%94%D1%95%D1%96%D1%97%D1%98%D1%99%D1%9A%D1%9B%D1%9C%D1%9E%D1%9F%D0%B0%D0%B1%D0%B2%D0%B3`

- Chromium: `ёђѓєѕіїјљњћќўџабвг`

With `encodeURI(window.location.hash)`:

- Firefox: `%25D1%2591%25D1%2592%25D1%2593%25D1%2594%25D1%2595%25D1%2596%25D1%2597%25D1%2598%25D1%2599%25D1%259A%25D1%259B%25D1%259C%25D1%259E%25D1%259F%25D0%25B0%25D0%25B1%25D0%25B2%25D0%25B3`

- Chromium: `%D1%91%D1%92%D1%93%D1%94%D1%95%D1%96%D1%97%D1%98%D1%99%D1%9A%D1%9B%D1%9C%D1%9E%D1%9F%D0%B0%D0%B1%D0%B2%D0%B3`

In other words, in Firefox it gets encoded twice. The alternative is to find a more cross-platform alternative to `window.location.hash` but `decodeURI` seems to work in the meantime.

**Comment by Blendify** [comment]
@jessetan

**Comment by jessetan** [comment]
@0giso can you check if PR #615 solves this for you?

**Comment by 0giso** [comment]
@jessetan I am not sure #615 can fix or not this problem, but I created test case to reproduce the issue.
https://github.com/0giso/testcase_for_sphinx_rtd_theme_issue495

It use sphinx_rtd_theme installed by pip.

**Comment by jessetan** [comment]
Thanks for the testcase. Can you explain the need for the unicode_ids extension? If I remove it, the TOC works correctly.

#615 does not influence the behavior you reported and is no longer necessary because Fx now behaves identical to Chrome.

**Comment by 0giso** [comment]
I need unicode_ids extension for non-English markdown document because if I don't use it, sphinx with recommonmark ignore non-English character in section title, then section hyper link url ends with '#' (no url fragment appended).

I [updated the test case](https://github.com/0giso/testcase_for_sphinx_rtd_theme_issue495/tree/832d63f42ce74f1f798d134fb8cda6ea26c1775b) as below to show unicode_ids necessity.

- Disable unicode_ids
- Section headers are translated from English to Japanese, Thai, or Korean.

Thank you.
